app/models/user.rb

```diff
@@ -544,4 +544,14 @@ def map_openid_registration(registration)
     self.email = registration['email'] if email.blank?
     self.username = registration['nickname'] if username.blank?
   end
+
+  def notes_for_tags(tagnames)
+    nids = NodeTag.includes(:node, :tag)
+    .references(:term_data)
+    .where(type: 'note')
+    .where(name: tagnames, uid: :uid)
+    @pagy, @notes = pagy(Node
+      .order('nid DESC')
+      .where(nid: nids), items: 24)
+  end
```

Comment:
@KarishmaVanwari You added the method to the private methods.

Also, I'm not sure this code will work. I played around with it but it's not returning what it's supposed to.

Can you test this code and see if it works properly?
```
def notes_for_tags(tagnames)
    Node.includes(:node_tag, :tag)
    .where('term_data.name IN (?)', tagnames)
    .references(:term_data, :node_tag)
    .where('node.uid = ?', uid)
    .where(type: 'note')
    .order('node.nid DESC')
  end
```

Reply:
Hey @TildaDares, could you please brief me as to how do I check if the given piece of code works well?

Reply:
@KarishmaVanwari Have you able to test this?

Reply:
I'm on it @TildaDares. Apologies for the delay since I had been busy with my university exams.